aya-bpf-macros/src/expand.rs

```diff
@@ -217,16 +224,42 @@ impl Xdp {
                 ));
             }
         }
+        let map = match pop_arg(&mut args, "map").as_deref() {
+            Some("cpumap") => Some(XdpMap::CpuMap),
+            Some("devmap") => Some(XdpMap::DevMap),
+            Some(_) => {
+                return Err(Error::new_spanned(
+                    "map",
+                    "invalid value. should be 'cpumap' or 'devmap'",
+                ))
+            }
+            None => None,
+        };
+
         err_on_unknown_args(&args)?;
-        Ok(Xdp { item, name, frags })
+        Ok(Xdp {
+            item,
+            name,
+            frags,
+            map,
+        })
     }
 
     pub fn expand(&self) -> Result<TokenStream> {
-        let section_prefix = if self.frags { "xdp.frags" } else { "xdp" };
+        let (prefix, delimiter) = if self.frags {
+            ("xdp.frags", "/")
+        } else {
+            ("xdp", "_")
```

Comment:
Delimiter in both cases should be `/`. See: https://docs.kernel.org/bpf/libbpf/program_types.html